Replace copy commit SHA with copy commit subject on the y s keybind in the commits view

karimkhaleel opened the pull request:
- **PR Description**

This changes the `copyCommitSHAToClipboard` into the `copyCommitSubjectToClipboard`. I am a user of the `<c-o>` keybind for copying commit hashes, so I fall into the category of people that won't miss the old keybinding.

We could just take the first commit of this PR and have no option to go back to the old `y s` keybind, or we can use the second commit (messy in my opinion, would want some feedback on a better way to do this) which would allow people to go back to the old keybind. 

Changing this keybinding was discussed in this issue: https://github.com/jesseduffield/lazygit/issues/3151

- **Please check if the PR fulfills these requirements**

* [x] Cheatsheets are up-to-date (run `go generate ./...`)
* [x] Code has been formatted (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#code-formatting))
* [ ] Tests have been added/updated (see [here](https://github.com/jesseduffield/lazygit/blob/master/pkg/integration/README.md) for the integration test guide) (not sure if we need a test here)
* [x] Text is internationalised (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#internationalisation)) (just fixed up the Russian and English internationalizations)
*  [x] Docs (specifically `docs/Config.md`) have been updated if necessary (I don't think anything needs updating here)
* [x] You've read through your own file changes for silly mistakes etc

<!--
Be sure to name your PR with an imperative e.g. 'Add worktrees view'
see https://github.com/jesseduffield/lazygit/releases/tag/v0.40.0 for examples
-->

stefanhaller commented:
We could also consider keeping the "Commit SHA" entry in the menu, at first position, but without a key binding. This way, at least people who have `y enter` in their muscle memory can continue to use that.

karimkhaleel commented:
> We could also consider keeping the "Commit SHA" entry in the menu, at first position, but without a key binding. This way, at least people who have `y enter` in their muscle memory can continue to use that.

This also works for me. Pushed a third commit that does that instead: 070b8493a3cecdd3300f397a7c70514310dc2e61

stefanhaller reviewed (commented):
Looks pretty good to me already. I think you should squash all commits into one now.

One concrete code suggestion inline below; apart from that, I have two thoughts about the menu:
1. I feel that "subject" and "message" should be next to each other, so the order should be either
```
  Commit SHA
s Commit subject
m Commit message
u Commit URL
d Commit diff
a Commit author
Cancel
```
or
```
  Commit SHA
u Commit URL
d Commit diff
s Commit subject
m Commit message
a Commit author
Cancel
```
I don't have a strong opinion, but I think I prefer the first one.

2. When seeing "message" next to "subject", maybe it's not totally clear what it means (just the body, or the full message?). I wonder if we should rename it to "Full commit message" or something like that?

karimkhaleel commented on pkg/commands/git_commands/commit.go:
```diff
@@ -146,6 +146,19 @@ func (self *CommitCommands) GetCommitMessage(commitSha string) (string, error) {
+		Arg("--format=%s", "--max-count=1", commitSha).
+		ToArgv()
+
+	rawSubject, err := self.cmd.New(cmdArgs).DontLog().RunWithOutput()
+	subjectArray := strings.Split(strings.TrimSpace(rawSubject), "\n")
+	if len(subjectArray) != 2 {
+		return "", errors.New("unexpected git output")
+	}
+	return subjectArray[1], err
+}
+
```
Great point!

karimkhaleel commented:
>1. I feel that "subject" and "message" should be next to each other, so the order should be either

Yeah, it does make sense for them to be next to each other.

>2. When seeing "message" next to "subject", maybe it's not totally clear what it means (just the body, or the full message?). I wonder if we should rename it to "Full commit message" or something like that?

I am not sure if I like the `Full commit message` title for the current `Commit message` field. I think it implies that we are going to be copying the subject and message when we're just copying the body of the commit message. 

What about: 
`Commit subject`    → `Commit message title` or `Commit message subject`
`Commit message` → `Commit message body`

stefanhaller commented:
> I am not sure if I like the `Full commit message` title for the current `Commit message` field. I think it implies that we are going to be copying the subject and message when we're just copying the body of the commit message.

But we *are* copying both subject and body (and we should). I was actually wondering whether it would be worth adding a third "message body" entry that really only copies the body without the subject, but I think it isn't going to be used often enough to warrant cluttering the menu with it.

karimkhaleel commented:
>But we are copying both subject and body (and we should).

:facepalm: oops! I'll try to be more attentive next time.

Squashed the commits and changed `Commit message` to `Full commit message` in English and Russian.

stefanhaller commented:
Very good, I'm happy with this now. I'll hold off merging this to wait for Jesse to have a look; especially for the decision to change the `y s` key binding.

jesseduffield commented:
Just read through the thread. Some points:
* I don't see a need for an option to copy just the commit description without the subject, so we don't need to bother with that
* I agree about using 'Full commit message' as the label
* I agree about leaving the hash as the first option without a keybinding

As for muscle memory, yes it will suck if people were using 's' and now they copy the wrong thing, but the vast majority would have just been hitting enter, and it's very low-stakes.
